File: android/app/src/test/java/com/phicdy/totoanticipation/model/RakutenTotoInfoParserTest.java
package com.phicdy.totoanticipation.model;

import org.junit.Test;

import java.util.ArrayList;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class RakutenTotoInfoParserTest {

    @Test
    public void gameListReturns() {
        ArrayList<Game> expectedGames = new ArrayList<>();
        expectedGames.add(new Game("浦和レッズ", "北海道コンサドーレ札幌"));
        expectedGames.add(new Game("ヴァンフォーレ甲府", "セレッソ大阪"));
        expectedGames.add(new Game("サンフレッチェ広島", "ベガルタ仙台"));
        expectedGames.add(new Game("鹿島アントラーズ", "ジュビロ磐田"));
        expectedGames.add(new Game("柏レイソル", "横浜F・マリノス"));
        expectedGames.add(new Game("アルビレックス新潟", "FC東京"));
        expectedGames.add(new Game("サガン鳥栖", "ヴィッセル神戸"));
        expectedGames.add(new Game("名古屋グランパス", "レノファ山口FC"));
        expectedGames.add(new Game("横浜FC", "ジェフユナイテッド千葉"));
        expectedGames.add(new Game("京都サンガF.C.", "松本山雅FC"));
        expectedGames.add(new Game("愛媛FC", "V・ファーレン長崎"));
        expectedGames.add(new Game("東京ヴェルディ", "ザスパクサツ群馬"));
        expectedGames.add(new Game("FC町田ゼルビア", "徳島ヴォルティス"));
        RakutenTotoInfoParser parser = new RakutenTotoInfoParser();
        ArrayList<Game> actualGames = parser.games(TestRakutenTotoInfoPage.text);
        for (int i = 0; i < actualGames.size(); i++) {
            assertThat(actualGames.get(i).homeTeam, is(expectedGames.get(i).homeTeam));
            assertThat(actualGames.get(i).awayTeam, is(expectedGames.get(i).awayTeam));
        }
    }

    @Test
    public void EmptyListReturnsWhenEmptyHtml() {
        RakutenTotoInfoParser parser = new RakutenTotoInfoParser();
        assertThat(parser.games("").size(), is(0));
    }

    @Test
    public void EmptyListReturnsWhenInvalidHtml() {
        RakutenTotoInfoParser parser = new RakutenTotoInfoParser();
        assertThat(parser.games("<html><body>hoge</body></html>").size(), is(0));
    }
}

File: android/legacy/src/test/java/com/phicdy/totoanticipation/legacy/model/RakutenTotoInfoParserTest.kt
package com.phicdy.totoanticipation.legacy.model

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat
import org.junit.Test

class RakutenTotoInfoParserTest {

    @Test
    fun gameListReturns() {
        val expectedGames = arrayListOf<Game>().apply {
            add(Game("浦和", "札幌"))
            add(Game("甲府", "Ｃ大阪"))
            add(Game("広島", "仙台"))
            add(Game("鹿島", "磐田"))
            add(Game("柏", "横浜Ｍ"))
            add(Game("新潟", "Ｆ東京"))
            add(Game("鳥栖", "神戸"))
            add(Game("名古屋", "山口"))
            add(Game("横浜Ｃ", "千葉"))
            add(Game("京都", "松本"))
            add(Game("愛媛", "長崎"))
            add(Game("東京Ｖ", "群馬"))
            add(Game("町田", "徳島"))
        }
        val parser = RakutenTotoInfoParser()
        val actualGames = parser.games(TestRakutenTotoInfoPage.text)
        for (i in actualGames.indices) {
            assertThat(actualGames[i].homeTeam, `is`(expectedGames[i].homeTeam))
            assertThat(actualGames[i].awayTeam, `is`(expectedGames[i].awayTeam))
        }
    }

    @Test
    fun emptyListReturnsWhenEmptyHtml() {
        val parser = RakutenTotoInfoParser()
        assertThat(parser.games("").size, `is`(0))
    }

    @Test
    fun emptyListReturnsWhenInvalidHtml() {
        val parser = RakutenTotoInfoParser()
        assertThat(parser.games("<html><body>hoge</body></html>").size, `is`(0))
    }

    @Test
    fun endTimeReturns() {
        val parser = RakutenTotoInfoParser()
        val deadlineTime = parser.deadlineTime(TestRakutenTotoInfoPage.text)
        assertThat(deadlineTime, `is`("13:50"))
    }

    @Test
    fun emptyTimeReturnsWhenEmptyHtml() {
        val parser = RakutenTotoInfoParser()
        assertThat(parser.deadlineTime(""), `is`(""))
    }

    @Test
    fun emptyTimeReturnsWhenInvalidHtml() {
        val parser = RakutenTotoInfoParser()
        assertThat(parser.deadlineTime("<html><body>hoge</body></html>"), `is`(""))
    }

}
